### ISSUE TITLE
RET-2 — Delete the legacy data layer (`mace/data/` + vendored torch_geometric)

### ISSUE THREAD

**Issue by aacostadiaz** [ACEsuit/mace#1598]
**Depends on:** RET-1 (#1597), DATA-3 (#1571), INF-5 (#1554) · **Blocks:** RET-3 (#1599), RET-6 (#1602)

**Prerequisite:** the Phase 0 characterization suite is committed — goldens, fixtures and the single-source tolerance table under `tests/golden/`, characterization tests under `tests/unit/`. References of the form `P0-N` name that work.

**Context:** The legacy data layer is the second subsystem to be physically deleted in the capability-by-capability retirement of the frozen `mace/` package (order: energy → **data** → conversion CLIs → dipole/polar/magnetic → LAMMPS jit → god-module + launcher legacy branch). This ticket removes the legacy graph/parsing/collation stack — all **8** tracked files under `mace/data/` (`__init__.py`, `atomic_data.py`, `augmentation.py`, `hdf5_dataset.py`, `lmdb_dataset.py`, `neighborhood.py`, `padding_tools.py`, `utils.py`) and the **8** files of the **vendored** `mace/tools/torch_geometric/` (`README.md`, `__init__.py`, `batch.py`, `data.py`, `dataloader.py`, `dataset.py`, `seed.py`, `utils.py`) — once the v1 data path is the default. The v1 replacement already exists and is proven: `mace_core.data` (typed numpy `Configuration` + parsing, CORE-3 (#1557)) and the v1 graph construction (DATA-3 (#1571)) produce the flat-dict data↔model contract (`{node_attrs, edge_index, positions, batch, cell, rcell, shifts, ptr, unit_shifts, …}`) **without vendored torch_geometric**, with neighbor lists, batching and collation pinned bit-exact against the legacy `Collater` at build time. This is a **deletion-only PR**: zero new v1 logic, budgeted for the deletion, one parity-test degradation, and the manifest flip. It drives `git ls-files mace/` down by exactly 16 files.

**Interface & constraints:**

1. **Preconditions (must already hold before this PR opens).** The `data` capability is **default-`v1`** and has been through its deprecation window with a green live-parity job; RET-1 (#1597) has already deleted the legacy energy models, so no default code path constructs legacy graphs. The DATA-3 (#1571) v1 data layer is the sole data path behind `--engine v1` / `MACE_ENGINE=v1`.

2. **What replaces each deleted piece (all already shipped — do not build here):**

| Deleted (legacy) | v1 replacement |
|---|---|
| `mace/data/utils.py` `Configuration`, XYZ parsing | `mace_core.data` typed numpy `Configuration` + `read_configurations` (CORE-3 (#1557)) |
| `mace/data/neighborhood.py` `get_neighborhood` (incl. the extended-cell / physical-cell / patched-row regimes and the fourth `cell` return value) | DATA-3 (#1571) neighbor-list module (padded/static-shape option; brute-force O(N²) helper from P0-7) |
| `mace/data/atomic_data.py` `AtomicData` (incl. `rcell` derivation), `mace/tools/torch_geometric/` `Collater` | DATA-3 (#1571) flat-dict collation (`ptr`/`batch`/`shifts`/`unit_shifts`/`cell`/`rcell`, no torch_geometric) |
| `mace/data/{hdf5,lmdb}_dataset.py`, `padding_tools.py` | DATA-2 (#1570) HDF5/LMDB backends + DATA-3 (#1571) padding contract |
| `mace/data/augmentation.py` (`Random3DRotation` + `create_random_rotation_loader`) | the MAG-1 (#1584) SO(3) magmom transform, registered through TRN-2 (#1576) |

3. **Parity degradation (the one substantive edit).** The DATA-3 (#1571) parity test today compares the v1 data layer **in-process against the live legacy `Collater`/`neighborhood`**. Deleting legacy removes that oracle, so this test **degrades to the P0-7 committed value-goldens**: it asserts the v1 layer reproduces the P0-7 reference values directly — the brute-force minimum-image edge sets (incl. shifts) on the triclinic / cell-smaller-than-cutoff self-image / non-periodic cluster / mixed-PBC fixtures, **the returned-cell value on each of those fixtures** (extended for the fully aperiodic one, physical for the mixed-PBC one, patched row for the zero-vacuum slab), and the `AtomicData`-collation expectations (one-hot vs z-table, shifts tensors, two-graph batch → correct `batch`/`ptr`/edge offsets). No legacy import remains in the test after this change. **Goldens are read, never regenerated here.**

4. **Manifest-driven, minimal surface.** The PR touches only `capabilities.toml` (INF-5 (#1554)) plus the deletion: flip the `data` capability's state `v1-default → retired`, and remove the now-satisfied `xfail(strict=True)` fitness function that guarded the flip (once burned it is deleted, not left as a permanently-green xfail). CI path-filters and per-package coverage floors regenerate from the manifest — no hand-edited CI. No production `packages/**` code changes.

5. **Import hygiene at the boundary.** After deletion, nothing outside the still-frozen legacy tree may import `mace.data` or `mace.tools.torch_geometric`. Legacy consumers of the data layer that are still physically present (e.g. `mace/tools/train.py`, `multihead_tools.py`, `run_train_utils.py`, `mace/cli/`, `mace/calculators/mace.py`) belong to capabilities retired later (RET-3 (#1599)..RET-6 (#1602)) and to the launcher's legacy branch removed at RET-6 (#1602); their `--engine=legacy` data access stops resolving here. That is admissible **only** where the consuming capability is already default-`v1` past its window or its parity has already degraded to a frozen golden — a live-legacy parity job that still needs `mace/data` is a blocker, not something to delete around (see Review focus).

6. **`augmentation.py` is the double dependency, and it is the reason the `[magnetic]` extra must already be clean.** It imports the vendored `mace.tools.torch_geometric.dataloader` unconditionally (`mace/data/augmentation.py:6`) *and* the real `torch_geometric.transforms.BaseTransform` behind a `find_spec` guard with a stub fallback (`:8-21`), while `setup.cfg:80-82` declares `torch-geometric` in the `[magnetic]` extra. MAG-1 (#1584) removes both dependencies from the ported transform and drops `torch-geometric` from the extra; verify neither survives before deleting.

**Task:**
1. Delete `mace/data/` in full (all 8 tracked files, including `augmentation.py`) and the vendored `mace/tools/torch_geometric/` package (all 8 tracked files).
2. Degrade the DATA-3 (#1571) data-parity test from "in-process vs live legacy" to "vs the P0-7 committed value-goldens" (edge sets + shifts + returned cell on the four neighbor fixtures; collation `batch`/`ptr`/edge-offset expectations); confirm zero legacy imports remain in it.
3. In `capabilities.toml`: set `data` → `retired`; remove the corresponding `xfail(strict=True)` fitness function.
4. Confirm `import-linter` / the meta-lint show no `packages/**` or v1-test import of `mace.data` or `mace.tools.torch_geometric`, and that no still-live legacy parity job depends on the deleted modules.
5. Verify `git ls-files mace/` decreased by exactly the 16 deleted files (no additions, no unrelated edits).

**Out of scope:** any new v1 data logic (that is DATA-2 (#1570)/DATA-3 (#1571), already merged); deleting the `convert_*` CLIs / `cg_cueq_tools` (RET-3 (#1599)); deleting the remaining legacy consumers `mace/tools/train.py`, `multihead_tools.py`, `mace/cli/`, `mace/calculators/` (RET-3 (#1599)..RET-6 (#1602)); regenerating any P0-7 golden; flipping the `data` capability's *default* (that flip is the earlier decision this deletion presupposes).

**Acceptance criteria:**
- [ ] `mace/data/` and `mace/tools/torch_geometric/` are physically removed; `git ls-files mace/` decreased by exactly those 16 files and nothing else changed under `mace/`.
- [ ] The DATA-3 (#1571) data-parity test passes against the P0-7 value-goldens — including the returned-cell value on all four neighbor fixtures — with no import of `mace.data` / `mace.tools.torch_geometric`.
- [ ] `capabilities.toml` shows `data = retired`; the guarding `xfail(strict=True)` fitness function is removed and the suite stays green.
- [ ] `import-linter` / meta-lint confirm no `packages/**` or v1 test imports the deleted modules; no still-live legacy parity job depends on them.
- [ ] **External importers of the vendored `mace/tools/torch_geometric` are confirmed absent** before deletion — one sweep over dependents / GitHub code search, recorded in the PR. It is a public-ish module inside an installed package, so "nobody in this repo imports it" is not the same as "nobody imports it".
- [ ] The in-tree complication is resolved too: `mace/data/augmentation.py`'s two torch_geometric dependencies (vendored dataloader + real `BaseTransform` with stub fallback) and the `torch-geometric` entry in the `[magnetic]` extra are all gone by MAG-1 (#1584) — verified, not assumed.
- [ ] CI path-filters and coverage floors regenerate cleanly from the manifest (no hand-edited CI in this PR); the PR contains zero new production logic.

**Verify:**
```bash
test -z "$(git ls-files mace/data mace/tools/torch_geometric)"          # deletion complete
python -m pytest tests/parity -m "not slow" -k data                     # degraded: v1 vs P0-7 value-goldens
python -m pytest packages/ -m "not slow"                                # v1 stack unaffected by the deletion
python -m pytest tests/golden -k "capabilities or inventory"            # manifest: data=retired, xfail burned
lint-imports                                                            # no import of mace.data / torch_geometric
```
**Review focus:** confirm the merge gate is honest that **no still-live legacy parity path imports `mace/data`** — RET-2 (#1598) lands before dipole/polar/magnetic (RET-4 (#1600)) and LAMMPS (RET-5 (#1601)) retire, and loading a live legacy model to compare against builds legacy graphs; if any such job still runs live-legacy, its inputs must already be sourced from frozen fixtures or it must degrade first, otherwise this deletion breaks it. Also verify this is a pure deletion (zero v1 logic added) and that the parity degradation reads P0-7 goldens — including the three cell regimes — rather than silently weakening the assertion.

---
